### PR TITLE
[FLOW-12] Add delete account feature

### PR DESCRIPTION
**Pull request by addison-ch**
- Added account deletion button and modal
- Depends on https://github.com/UWFlow/uwflow/pull/127 being merged as well
- FLOW-12

uwflowdeletedemo

### CODE REVIEW

File: src/utils/Api.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ const makeRequest = async <T extends object>(
   body?: { body: any },
 ): Promise<[T, number]> => {
   const res = await fetch(endpoint, {
-    mode: process.env.NODE_ENV === 'development' ? 'no-cors' : 'cors',
+    // mode: process.env.NODE_ENV === 'development' ? 'no-cors' : 'cors',
```
Comment (addison-ch) [inline src/utils/Api.tsx:8]: Setting 'no-cors' seems to make the request not add any auth headers

File: src/pages/profilePage/ProfileInfoHeader.tsx [hunk 2]
```diff
@@ -35,17 +35,25 @@ const ProfileInfoHeader = ({ user }: ProfileInfoHeaderProps) => {
         <UserInfoWrapper>
           <UserName>{user.full_name}</UserName>
           <UserProgram>{user.program}</UserProgram>
-          <UserEmailWrapper>
+          <UserProfileOptionWrapper>
```
Comment (edwinzhng) [inline src/pages/profilePage/ProfileInfoHeader.tsx:38]: Can you add screenshots?

Comment (addison-ch) [reply src/pages/profilePage/ProfileInfoHeader.tsx:38]: Added a GIF

File: src/components/delete/DeleteAccountModalContent.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      <ConfirmationText>
+        Are you sure you want to continue? This action cannot be undone.
+      </ConfirmationText>
+      <ButtonsWrapper>
+        <Button
+          color={theme.dark3}
+          handleClick={onRequestClose}
+          margin="0 16px 0 0"
+        >
+          Cancel
```
Comment (edwinzhng) [inline src/components/delete/DeleteAccountModalContent.tsx:69]:
```suggestion
          <LightText>Cancel</LightText>
```

File: src/components/delete/DeleteAccountModalContent.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        </Button>
+        <Button color={theme.red} handleClick={handleDelete} disabled={loading}>
+          {loading ? 'Deleting...' : 'Delete'}
+        </Button>
```
Comment (edwinzhng) [inline src/components/delete/DeleteAccountModalContent.tsx:73]:
```suggestion
        <Button color={theme.red} handleClick={handleDelete} loading={loading}>
          <LightText>Delete</LightText>
        </Button>
```

**Review by edwinzhng — COMMENTED** [review]
Can you try these to see how they look?

File: src/components/delete/DeleteAccountModalContent.tsx (hunk 3 above)
Comment (edwinzhng) [reply src/components/delete/DeleteAccountModalContent.tsx:69]: We do this in the `CourseReviewBox` too

File: src/components/delete/DeleteAccountModalContent.tsx (hunk 4 above)
Comment (edwinzhng) [reply src/components/delete/DeleteAccountModalContent.tsx:73]: Shouldn't need to manually handle loading, the spinner should show up in this case

File: src/components/delete/styles/DeleteAccount.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  margin-bottom: 24px;
+`;
+
+export const ButtonsWrapper = styled.div`
+  display: flex;
+  justify-content: flex-end;
+`;
+
+export const ConfirmationText = styled.div`
+  margin-bottom: 16px;
```
Comment (edwinzhng) [inline src/components/delete/styles/DeleteAccount.tsx:28]:
```suggestion
  margin-bottom: 40px;
```

File: src/components/delete/styles/DeleteAccount.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  background: white;
+  width: 400px;
+  border-radius: 4px;
+  max-width: 95vw;
+  min-width: 300px;
+`;
+
+export const DeleteAccountTitle = styled.div`
+  ${Heading2}
+  margin-bottom: 24px;
```
Comment (edwinzhng) [inline src/components/delete/styles/DeleteAccount.tsx:19]:
```suggestion
  margin-bottom: 40px;
```

**Pull request merged**